### PR TITLE
WIP: add new state machines

### PR DESCRIPTION
**Pull request by LucienMorey**

### CODE REVIEW

File: controllers/acquire_cone.py [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+        if self.gripper.get_full_closed():
+            self.done()
+
+    def target_left(self) -> None:
+        self.targeting_left = True
+
+    def target_right(self) -> None:
+        self.targeting_left = False
+
+    def get_cone_pickup(self) -> tuple[Pose2d, Rotation2d]:
```
Comment (james-ward) [inline controllers/acquire_cone.py:76]: @outsidermm This function should not live in the state machine file. All field geometry calculations should live in their own file amongst their own kind. ;)

File: controllers/arm.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+    def go_to_setpoint(self, setpoint: Setpoint) -> None:
+        # Only restart the state machine if the setpoint is different
+        if setpoint != self._target_setpoint:
+            self.done()
+            self._target_setpoint = setpoint
+            self.engage()
+
+    def get_angle(self) -> float:
+        return self.arm_component.get_angle()
```
Comment (LucienMorey) [inline controllers/arm.py:20]:
```suggestion
        return self.arm.get_angle()
```

File: controllers/arm.py [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+    def at_goal(self) -> bool:
+        return not self.is_executing
+
+    @state(first=True, must_finish=True)
+    def retract_arm(self) -> None:
+        self.next_state("rotate_arm")
```
Comment (LucienMorey) [inline controllers/arm.py:27]:
```suggestion
    def retracting_arm(self) -> None:
        self.next_state("rotating_arm")
```
Dont we generally try to name all of our states with "ing" so we know they are doing something and its not part of the interface to tell it to do it

Comment (james-ward) [reply controllers/arm.py:27]: Yep!

File: utilities/game.py [hunk 4]
```diff
@@ -146,3 +185,35 @@ def get_staged_pieces(alliance: wpilib.DriverStation.Alliance) -> list[Translati
...
+    return get_team() == wpilib.DriverStation.Alliance.kRed
+
+
+def get_team() -> wpilib.DriverStation.Alliance:
+    return wpilib.DriverStation.getAlliance()
+
+
+def get_cone_pickup(targeting_left: bool) -> tuple[Pose2d, Rotation2d]:
+    # if we want the substation to be as if we are on the red alliance
+    red_side = is_red() != swap_substation
```
Comment (auscompgeek) [inline utilities/game.py:200]: This will always be True, as a tunable instance is never equal to a boolean…

Comment (james-ward) [reply utilities/game.py:200]: I think more broadly that file needs a really good refactor - it's not very clear how it all works, and there are a lot of arguments to functions that probably aren't necessary.

**Review by OliverW10 — APPROVED** [review]
Merging so we can work off of main

**Review by LucienMorey — COMMENTED** [review]
LGTM but I cant approve this since its originally mine

**Pull request merged**